packages/react-components/react-table/src/hooks/useSelection.ts

```diff
@@ -1,38 +1,70 @@
 import * as React from 'react';
 import { useEventCallback, usePrevious } from '@fluentui/react-utilities';
 import { createSelectionManager } from './selectionManager';
-import { GetRowIdInternal, RowId, SelectionMode, SelectionStateInternal } from './types';
+import { GetRowIdInternal, RowId, SelectionMode, SelectionStateInternal, UseTableSelectionOptions } from './types';
+
+interface UseSelectionOptions<TItem> {
+  selectionMode: SelectionMode;
+  items: TItem[];
+  getRowId: GetRowIdInternal<TItem>;
+  onRowSelectionChange?: UseTableSelectionOptions['onRowSelectionChange'];
+  defaultSelectedRows?: RowId[];
+}
+
+export function useSelection<TItem>(options: UseSelectionOptions<TItem>): SelectionStateInternal {
+  const { selectionMode, items, getRowId, onRowSelectionChange = () => undefined, defaultSelectedRows } = options;
 
-export function useSelection<TItem>(
-  selectionMode: SelectionMode,
-  items: TItem[],
-  getRowId: GetRowIdInternal<TItem>,
-): SelectionStateInternal {
   const prevSelectionMode = usePrevious(selectionMode);
-  const [selected, setSelected] = React.useState(() => new Set<RowId>());
```

Comment:
removed the react state here, there's actually no point in duplicating the same information that is in the selection manager. Instead added a force updater to update every time the selection is updated